### DISCUSSION TITLE
How to show module titles when using with Pages?

### DISCUSSION THREAD

**Discussion by manojLondhe** [joomlatools/joomlatools-pages#786] [Q&A] [answered]
Hi,

Wanted to show module title as set in module settings as when using with Pages.

Tried both

`<ktml:modules position="pages-footer-2" params="<?= json(['showtitle' => 1], true); ?>">`

and

`<ktml:modules position="pages-footer-2" showtitle="1">`

But it does not show module title in o/p, shows only module content,

**Comment by johanjanssens** [comment, marked as answer]
### 1. params

The `<ktml:modules ... >` tag doesn’t support `params=“<?= json([‘showtitle’ => 1], true); ?>”` Params only exists for a module and are defined when you instantiate the module. 

The right syntax would be:

```html
<ktml:modules position="pages-footer-2" showtitle="1">
```

### 2. showtitle

A `title` in Joomla is not rendered by the module. It’s rendered by the module chrome. See: https://github.com/joomla/joomla-cms/blob/staging/templates/system/html/modules.php

You can pass the info to chrome in as a style attribute, for example:

```html
<ktml:modules position="position-2" style="xhtml">
```

This will render the title. This works in the same way as the `jdoc:include` counterpart in Joomla.

```html
<jdoc:include type="modules" name="position-2" style="xhtml" />
```

The additional attributes you add to `ktml:modules` are injected as `attribs` when rendering the module, in the same way as Joomla does, you can use them in the chrome function or in the `onRenderModule` and `onAfterRenderModule`  events in Joomla.

There is no way right now to override the `showtitle` module property for all modules when rendering  through `ktml:modules`, the settings of the individual modules apply.  

I don't think it makes much sense to make that possiblity, the only module property that this could works for is `showtitle` , however you can easily decide to not render any titles by not setting any chrome, which solves the problem.
